aspnetcore/tutorials/signalr-typescript-webpack.md

```diff
@@ -24,10 +24,247 @@ In this tutorial, you learn how to:
+
+1. Create *src/index.ts* with the following content:
+
+    [!code-typescript[index.ts](signalr-typescript-webpack/sample/3.x/snippets/index1.ts?name=snippet_IndexTsPhase1File)]
+
+    The preceding TypeScript retrieves references to DOM elements and attaches two event handlers:
+
+    * `keyup`: This event fires when the user types something in the textbox identified as `tbMessage`. The `send` function is called when the user presses the **Enter** key.
+    * `click`: This event fires when the user clicks the **Send** button. The `send` function is called.
+
+## Configure the ASP.NET Core app
+
+1. In the `Startup.Configure` method, add calls to [UseDefaultFiles](/dotnet/api/microsoft.aspnetcore.builder.defaultfilesextensions.usedefaultfiles#Microsoft_AspNetCore_Builder_DefaultFilesExtensions_UseDefaultFiles_Microsoft_AspNetCore_Builder_IApplicationBuilder_) and [UseStaticFiles](/dotnet/api/microsoft.aspnetcore.builder.staticfileextensions.usestaticfiles#Microsoft_AspNetCore_Builder_StaticFileExtensions_UseStaticFiles_Microsoft_AspNetCore_Builder_IApplicationBuilder_).
+
+   [!code-csharp[Startup](signalr-typescript-webpack/sample/3.x/Startup.cs?name=snippet_UseStaticDefaultFiles&highlight=2-3)]
+
+   The preceding code allows the server to locate and serve the *index.html* file, whether the user enters its full URL or the root URL of the web app.
+
+1. At the end of the `Startup.Configure` method, map a */hub* route to the `ChatHub` hub. Replace the code that displays *Hello World!* with the following line: 
+
+   [!code-csharp[Startup](signalr-typescript-webpack/sample/3.x/Startup.cs?name=snippet_UseSignalR&highlight=3)]
+
+1. In the `Startup.ConfigureServices` method, call [AddSignalR](/dotnet/api/microsoft.extensions.dependencyinjection.signalrdependencyinjectionextensions.addsignalr#Microsoft_Extensions_DependencyInjection_SignalRDependencyInjectionExtensions_AddSignalR_Microsoft_Extensions_DependencyInjection_IServiceCollection_). It adds the SignalR services to your project.
+
+   [!code-csharp[Startup](signalr-typescript-webpack/sample/3.x/Startup.cs?name=snippet_AddSignalR)]
+
+1. Create a new directory, called *Hubs*, in the project root. Its purpose is to store the SignalR hub, which is created in the next step.
+
+1. Create hub *Hubs/ChatHub.cs* with the following code:
+
+    [!code-csharp[ChatHub](signalr-typescript-webpack/sample/3.x/snippets/ChatHub.cs?name=snippet_ChatHubStubClass)]
+
+1. Add the following code at the top of the *Startup.cs* file to resolve the `ChatHub` reference:
+
+    [!code-csharp[Startup](signalr-typescript-webpack/sample/3.x/Startup.cs?name=snippet_HubsNamespace)]
+
+## Enable client and server communication
+
+The app currently displays a simple form to send messages. Nothing happens when you try to do so. The server is listening to a specific route but does nothing with sent messages.
+
+1. Execute the following command at the project root:
+
+    ```console
+    npm install @aspnet/signalr
```

Comment:
I haven't heard since I did [the JS tutorial update](https://github.com/aspnet/AspNetCore.Docs/pull/13101#issuecomment-509715474).  

cc @anurse